frontend/src/lib/validation.ts

```diff
@@ -211,3 +211,12 @@ export function validateBareMetalOSImageURL(value: string, t: TFunction) {
 
     return undefined
 }
+
+export function validateURL(url: string, t: TFunction) {
+    const regex = /(http|https):\/\/(www.)?(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?((\/|\/)?([\w#!:.?+=&%@!\-/]))\S*$/
```

Comment:
This regex looks very complex and it seems to be checking more than that it's just a syntactically correct URL. For example, it only allows http or https protocol... so if this is specific to what an Ansible URL should like, let's not use a generic name.

Other concerns with it:
1. Does not begin with `^` - can invalid stuff be inserted at the beginning?
2. Why do we check for optional `www.`? I think we can just leave that out.
3. The character class `\w` includes underscore, which is not valid in the domain part of a URL.

I tried with `!3kfj   http://hello_you.com` and it was accepted.

Reply:
Maybe there's a library we can use to try parsing the URL, then if successful check the protocol and any other restrictions we need to place for an Ansible URL.